xlators/system/posix-acl/src/posix-acl.c

```diff
@@ -468,71 +480,53 @@ __posix_acl_set_specific(inode_t *inode, xlator_t *this, gf_boolean_t is_access,
     return ret;
 }
 
-int
+static void
 __posix_acl_set(inode_t *inode, xlator_t *this, struct posix_acl *acl_access,
                 struct posix_acl *acl_default)
 {
-    int ret = 0;
     struct posix_acl_ctx *ctx = NULL;
 
     ctx = posix_acl_ctx_get(inode, this);
     if (!ctx)
-        goto out;
+        return;
 
     ctx->acl_access = acl_access;
     ctx->acl_default = acl_default;
-
-out:
-    return ret;
 }
 
-int
+static void
 __posix_acl_get(inode_t *inode, xlator_t *this, struct posix_acl **acl_access_p,
                 struct posix_acl **acl_default_p)
 {
-    int ret = 0;
     struct posix_acl_ctx *ctx = NULL;
 
     ctx = posix_acl_ctx_get(inode, this);
     if (!ctx)
-        goto out;
+        return;
```

Comment:
Probably this was also a bug. How the returned code was handled by some callers also seems a bug.

Reply:
Why? They seem to check that the value of acl_access_p and/or acl_default_p is non NULL, isn't it sufficient?

Reply:
I'm not sure if a NULL ACL is always interpreted as an error by the code. I think it's ok to have a missing (=NULL) ACL and, if that's the case, it's not the same a NULL ACL than an inaccessible ACL.